analysers/analyser_osmosis_relation_public_transport.py

```diff
@@ -366,6 +368,144 @@
+                return null;
+      end case;
+    end loop;
+   return full_way;
+end;
+$$;
+"""
+
+sql101 = """
+CREATE TEMP TABLE route_linestring AS
+SELECT id,
+       relations.tags->'route' AS public_transport_mode,
+       ST_Transform(generate_linestring_geom(id), {0}) AS geom
+FROM relations
+WHERE relations.tags->'type' = 'route'
+  AND relations.tags->'route' IN ('train',
+                             'subway',
+                             'monorail',
+                             'tram',
+                             'bus',
+                             'trolleybus',
+                             'aerialway',
+                             'ferry',
+                             'coach',
+                             'funicular',
+                             'share_taxi',
+                             'light_rail',
+                             'school_bus')
+  AND (NOT relations.tags?(relations.tags->'route')
+       OR relations.tags->(relations.tags->'route') != 'on_demand')
+"""
+
+sql101b = """
+CREATE INDEX route_linestring_idx ON route_linestring USING gist(geom)
+"""
+
+sql102 = """
+CREATE TEMP TABLE platform_that_can_project AS
+(
+SELECT route_linestring.id AS route_id,
+       stop_platform.member_type || stop_platform.mid AS stop_id,
+       stop_platform.geom AS stop,
+       ROW_NUMBER () OVER (PARTITION BY route_linestring.id
+                     ORDER BY stop_platform.morder) AS stop_order,
+       ROW_NUMBER () OVER (PARTITION BY route_linestring.id
+                     ORDER BY ST_LineLocatePoint(ST_OffsetCurve(route_linestring.geom, -10), stop_platform.geom) DESC) AS projected_stop_order
+FROM stop_platform
+JOIN route_linestring ON route_linestring.id = stop_platform.id
+WHERE stop_platform.mrole in ('platform',
+                              'platform_exit_only',
+                              'platform_entry_only')
+  AND stop_platform.member_type='N'
+  AND GeometryType(ST_OffsetCurve(route_linestring.geom, -10)) = 'LINESTRING'
+  AND ST_LineLocatePoint(route_linestring.geom, ST_ClosestPoint(route_linestring.geom, stop_platform.geom)) <> 1
+)
+"""
+
+sql102b = """
+CREATE INDEX platform_that_can_project_idx ON platform_that_can_project USING gist(stop)
+"""
+
+sql103 = """
+SELECT DISTINCT ON (route_id)
+  platform_that_can_project.route_id,
+  ST_AsText(ST_Transform(platform_that_can_project.stop,4326)) AS geom
+FROM platform_that_can_project
+WHERE stop_order <> projected_stop_order
+"""
+
+sql110 = """
+SELECT
+  platform_that_can_project.route_id,
+  platform_that_can_project.stop_id,
+  ST_AsText(ST_Transform(platform_that_can_project.stop, 4326))
+FROM platform_that_can_project
+JOIN route_linestring ON route_linestring.id = platform_that_can_project.route_id
+JOIN (
+  SELECT
+    platform_that_can_project.route_id AS route_id,
+    MAX(platform_that_can_project.stop_order) AS last_platform
+  FROM platform_that_can_project
+  GROUP BY platform_that_can_project.route_id
```

Comment:
Maybe you can precompute the last platform directly in `platform_that_can_project` when window function `lead()` is null.